### PR TITLE
Implement Pylint's `too-many-arguments` rule (`PLR0913`)

### PR DESCRIPTION
**Pull request by akhildevelops**
Pylint Rule: too-many-arguments / R0913 https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-many-arguments.html

related to https://github.com/charliermarsh/ruff/issues/970

### CODE REVIEW

File: src/rules/pylint/rules/too_many_args.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violations;
+use rustpython_ast::{Arguments, Stmt};
+
+/// PLR0913
+pub fn too_many_args(checker: &mut Checker, args: &Arguments, stmt: &Stmt) {
+    if args.args.len() > checker.settings.pylint.max_args {
```
Comment (charliermarsh) [inline src/rules/pylint/rules/too_many_args.rs:9]: I suspect that this also needs to include `kwargs`, `kwonly_args`, and the other fields on `Arguments`. Can you check?

File: src/rules/pylint/settings.rs [hunk 2]
```diff
@@ -53,17 +53,28 @@ pub struct Options {
...
     /// Constant types to ignore when used as "magic values".
     pub allow_magic_value_types: Option<Vec<ConstantType>>,
+    #[option(
+        default = r#"5"#,
+        value_type = "usize",
+        example = r#"
+            allow-magic-value-types = 5
+        "#
+    )]
+    /// Constant types to ignore when used as "magic values".
```
Comment (charliermarsh) [inline src/rules/pylint/settings.rs:63]: Looks like this was copied over from `allow_magic_value_types` -- mind writing something relevant for this new option?

File: src/rules/pylint/settings.rs [hunk 3]
```diff
@@ -53,17 +53,28 @@ pub struct Options {
     )]
     /// Constant types to ignore when used as "magic values".
     pub allow_magic_value_types: Option<Vec<ConstantType>>,
+    #[option(
+        default = r#"5"#,
+        value_type = "usize",
+        example = r#"
+            allow-magic-value-types = 5
```
Comment (charliermarsh) [inline src/rules/pylint/settings.rs:60]: Nit: should be `max-args = 5`

File: src/violations.rs [hunk 4]
```diff
@@ -870,6 +870,21 @@ impl Violation for GlobalVariableNotAssigned {
...
+    }
+);
+
+impl Violation for TooManyArgs {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooManyArgs { c_args, max_args } = self;
+        format!("Too many arguments ({c_args}/{max_args})")
+    }
+}
```
Comment (charliermarsh) [inline src/violations.rs:886]: Can you move this into `src/rules/pylint/rules/too_many_args.rs`? We're trying to colocate violations with their rule implementations going forward.

File: src/rules/pylint/rules/too_many_args.rs (hunk 1 above)
Comment (akhildevelops) [reply src/rules/pylint/rules/too_many_args.rs:9]: This rule only checks for args: https://github.com/PyCQA/pylint/blob/af555eddf6ecdb1993345dce728a6b4c38dd4a02/pylint/checkers/design_analysis.py#L519-L524

**Comment by akhildevelops** [comment]
Hi @charliermarsh :headphones: 

I've made changes as suggested and also added an extra configuration param: [`ignored-argument-names`](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#ignored-argument-names). It's used to [ignore arguments that match the pattern](https://github.com/PyCQA/pylint/blob/af555eddf6ecdb1993345dce728a6b4c38dd4a02/pylint/checkers/design_analysis.py#L511-L518) and then validate the rule.

File: src/rules/pylint/rules/too_many_args.rs (hunk 1 above)
Comment (charliermarsh) [reply src/rules/pylint/rules/too_many_args.rs]: Ohh, interesting! Thanks.

File: resources/test/fixtures/pylint/too_many_args.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+def foo(x,y,z,_t,_u,_v,_w,r): # OK _.* can be ignored as per --ignored-argument-names configuration
+    pass
+
+def foo(x,y,z,ignored_u,unused_v,r): #OK ^ignored_|^unused_ can be ignored
+    pass
+
+def foo(x,y,z,u_ignored,v_unused,r): # Too many arguments (6/5)
+    pass
+
+def foo(x,y,z,skip_u,skip_v,r): # Too many arguments (6/5)
```
Comment (charliermarsh) [inline resources/test/fixtures/pylint/too_many_args.py:19]: Can we add some test cases here for: arguments with default values (`a=1`), keyword-only args (`def f(a, b, *, c, d)`), and positional-only args (`def f(a, b, /, c, d)`)?

File: src/rules/pylint/settings.rs [hunk 6]
```diff
@@ -53,27 +54,52 @@ pub struct Options {
...
+    #[option(default = r"5", value_type = "usize", example = r"max_args = 5")]
+    /// Maximum number of arguments for function / method.
+    pub max_args: Option<usize>,
+    #[option(
+        default = r"^_.*|^ignored_|^unused_",
+        value_type = "String",
+        example = r"ignored-argument-names = skip_.*"
+    )]
+    /// Argument names that match this expression will be ignored.
+    pub ignored_argument_names: Option<String>,
```
Comment (charliermarsh) [inline src/rules/pylint/settings.rs:66]: Definitely appreciate you going through the exercise of adding this, but could we instead use `settings.dummy_variable_rgx`? It's kind of nice to have a centralized definition for "ignorable names", and we already use that across a variety of checks.

Comment (akhildevelops) [reply src/rules/pylint/settings.rs:66]: Sure! I will use this. Thanks for pointing out :point_up:

Comment (charliermarsh) [reply src/rules/pylint/settings.rs:66]: No prob! Sorry that you went through the effort of doing this only to roll it back. But hopefully it'll be a more cohesive experience for users.

**Comment by akhildevelops** [comment]
Hi @charliermarsh, Done with the changes.

**Pull request merged**